chips/nrf52/src/uart.rs

```diff
@@ -1,22 +1,156 @@
+    pub task_startrx: WriteOnly<u32, Task::Register>, // 0x000
+    pub task_stoprx: WriteOnly<u32, Task::Register>,  // 0x004
+    pub task_starttx: WriteOnly<u32, Task::Register>, // 0x008
+    pub task_stoptx: WriteOnly<u32, Task::Register>,  // 0x00c
+    _reserved1: [u32; 7],                             // 0x010-0x02c
+    pub task_flush_rx: WriteOnly<u32, Task::Register>, // 0x02c
+    _reserved2: [u32; 52],                            // 0x030-0x100
+    pub event_cts: ReadWrite<u32, Event::Register>,   // 0x100-0x104
+    pub event_ncts: ReadWrite<u32, Event::Register>,  // 0x104-0x108
+    _reserved3: [u32; 2],                             // 0x108-0x110
+    pub event_endrx: ReadWrite<u32, Event::Register>, // 0x110-0x114
+    _reserved4: [u32; 3],                             // 0x114-0x120
+    pub event_endtx: ReadWrite<u32, Event::Register>, // 0x120-0x124
+    pub event_error: ReadWrite<u32, Event::Register>, // 0x124-0x128
+    _reserved6: [u32; 7],                             // 0x128-0x144
+    pub event_rxto: ReadWrite<u32, Event::Register>,  // 0x144-0x148
+    _reserved7: [u32; 1],                             // 0x148-0x14C
+    pub event_rxstarted: ReadWrite<u32, Event::Register>, // 0x14C-0x150
+    pub event_txstarted: ReadWrite<u32, Event::Register>, // 0x150-0x154
+    _reserved8: [u32; 1],                             // 0x154-0x158
+    pub event_txstopped: ReadWrite<u32, Event::Register>, // 0x158-0x15c
+    _reserved9: [u32; 41],                            // 0x15c-0x200
+    pub shorts: ReadWrite<u32, Shorts::Register>,     // 0x200-0x204
+    _reserved10: [u32; 64],                           // 0x204-0x304
+    pub intenset: ReadWrite<u32, Interrupt::Register>, // 0x304-0x308
+    pub intenclr: ReadWrite<u32, Interrupt::Register>, // 0x308-0x30C
+    _reserved11: [u32; 93],                           // 0x30C-0x480
+    pub errorsrc: ReadWrite<u32, ErrorSrc::Register>, // 0x480-0x484
+    _reserved12: [u32; 31],                           // 0x484-0x500
+    pub enable: ReadWrite<u32, Uart::Register>,       // 0x500-0x504
+    _reserved13: [u32; 1],                            // 0x504-0x508
+    pub pselrts: ReadWrite<u32, Psel::Register>,      // 0x508-0x50c
+    pub pseltxd: ReadWrite<u32, Psel::Register>,      // 0x50c-0x510
+    pub pselcts: ReadWrite<u32, Psel::Register>,      // 0x510-0x514
+    pub pselrxd: ReadWrite<u32, Psel::Register>,      // 0x514-0x518
+    _reserved14: [u32; 3],                            // 0x518-0x524
+    pub baudrate: ReadWrite<u32, Baudrate::Register>, // 0x524-0x528
+    _reserved15: [u32; 3],                            // 0x528-0x534
+    pub rxd_ptr: ReadWrite<u32, Pointer::Register>,   // 0x534-0x538
+    pub rxd_maxcnt: ReadWrite<u32, Counter::Register>, // 0x538-0x53c
+    pub rxd_amount: ReadOnly<u32, Counter::Register>, // 0x53c-0x540
+    _reserved16: [u32; 1],                            // 0x540-0x544
+    pub txd_ptr: ReadWrite<u32, Pointer::Register>,   // 0x544-0x548
```

Comment:
This really ought to be something like `ReadWrite<&'static u8, Pointer::Register>`. Is that possible?

Reply:
No, I don't think that is possible because `ÌntLike` doesn't support references yet but I think it would be nice to have such a feature though :+1: 

It would potentially remove some ugly syntax to cast ```*const u8 as u32``` all the time when using DMA.

I guess we need something like:
```rust
impl<'a> IntLike for &'a u8 {
    fn zero() -> Self {
        0
    }
}
```

Reply:
Here's an example of implementing IntLike: https://github.com/helena-project/tock/compare/sam4l-new-regs-dma-full?expand=1#diff-4921d4a2c624f232e8f2e7cd92d70a00R654